### PR TITLE
Closes #12 `derive_param_computed`

### LINKED ISSUE #12: A value level validation use case, where `admiral` is used to validate a small chunk of the data:
A value level validation use case, where `admiral` is used to validate a small chunk of the data:

* validate a parameter by using [`admiral::derive_param_computed`](https://pharmaverse.github.io/admiral/cran-release/reference/derive_param_computed.html).
* Derivation text: derive new parameter/records for `ADEX.PARAMCD = ADPCYMG` (Actual Dose per Cycle) based on existing parameters - Actual Dose per Cycle (ADPCYMG) = Total Amount of Dose (TOTDOSE) / Number of Cycles (NUMCYC);
* `admiral::derive_param_computed` can add a parameter computed from the analysis value of other parameters.
In this example, ` ADEX.AVAL (where ADEX.PARAPMCD = ADPCYMG)` 
`= ADEX.AVAL (where ADEX.PARAPMCD = TOTDOSE) / ADEX.AVAL (where ADEX.PARAPMCD = NUMCYC)`

```r
# read in data
path_ads <- "xxxx"
adex <- haven::read_sas(paste0(path_ads, "adex.sas7bdat")) %>% 
  select(USUBJID, PARAM, PARAMCD, AVAL)

# subset input data to PARAMCD = "TOTDOSE", "NUMCYC"
adex_TOTDOSE_NUMCYC <- adex %>% filter(PARAMCD %in% c("TOTDOSE", "NUMCYC"))
# admiral::derive_param_computed()
val <- 
  derive_param_computed(
  adex_TOTDOSE_NUMCYC,
  by_vars = exprs(USUBJID),
  parameters = c("TOTDOSE", "NUMCYC"),
  analysis_value = AVAL.TOTDOSE / AVAL.NUMCYC,
  set_values_to = exprs(
    PARAMCD = "ADPCYMG",
    PARAM = "Actual Dose per Cycle (mg)"
  )
) 

# compare with primary
diffdf::diffdf(adex %>% filter(PARAMCD == "ADPCYMG"), 
               val %>% filter(PARAMCD == "ADPCYMG"), 
               keys = c("USUBJID", "PARAMCD"))
```

_Originally posted by @kaz462 in https://github.com/pharmaverse/comms_wg/issues/2#issuecomment-1496730441_

### PR DESCRIPTION
**Pull request by kaz462**
1. added some folders and a blog post `posts/admiral/valuelevel/derive_param_computed.qmd`
2. updated the UI, see: https://kaz462.github.io/comms_wg_/

### CODE REVIEW

File: posts/admiral/valuelevel/derive_param_computed.qmd [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+options(width = narrow)
+```
+
+<!--------------- post begins here ----------------->
+
+## Introduction
+
+When creating ADaM Basic Data Structure (BDS) datasets, we often encounter deriving a new parameter based on the analysis values (e.g., AVAL) of other parameters.
+
+The `admiral` function `derive_param_computed()` works like a calculator to derive new records without worrying about merging and combining datasets, all you need is a derivation formula, which also improves the readability of the code.
```
Comment (bms63) [inline posts/admiral/valuelevel/derive_param_computed.qmd:27]: Can these be made into links: admiral and derive_param_computed()

File: posts/admiral/valuelevel/derive_param_computed.qmd [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+In this example, `ADEX.AVAL when ADEX.PARAMCD = ADPCYMG` can be derived as:
+
+$$
+AVAL (PARAMCD = ADPCYMG) = \frac{AVAL (PARAMCD = TOTDOSE)}{AVAL (PARAMCD = NUMCYC)}
+$$
+
+``` r
+# read in data
+path_ads <- ""
+adex <- haven::read_sas(paste0(path_ads, "adex.sas7bdat")) %>%
```
Comment (bms63) [inline posts/admiral/valuelevel/derive_param_computed.qmd:48]: Apologies - you might of said this before - but is this data available?  Can we include at the end of the post?

File: posts/admiral/valuelevel/derive_param_computed.qmd [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  set_values_to = exprs(
+    PARAMCD = "ADPCYMG",
+    PARAM = "Actual Dose per Cycle (mg)"
+  )
+) 
+
+# compare with primary
+diffdf::diffdf(adex %>% filter(PARAMCD == "ADPCYMG"), 
+               val %>% filter(PARAMCD == "ADPCYMG"), 
+               keys = c("USUBJID", "PARAMCD"))
```
Comment (bms63) [inline posts/admiral/valuelevel/derive_param_computed.qmd:69]: We should mention diffdf and what it does in the Introduction

File: posts/admiral/valuelevel/derive_param_computed.qmd [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+``` r
+# read in data
+path_ads <- ""
+adex <- haven::read_sas(paste0(path_ads, "adex.sas7bdat")) %>% 
+  select(USUBJID, PARAM, PARAMCD, AVAL)
+
+# subset input data to PARAMCD = "TOTDOSE", "NUMCYC"
+adex_TOTDOSE_NUMCYC <- adex %>% filter(PARAMCD %in% c("TOTDOSE", "NUMCYC"))
+# admiral::derive_param_computed()
```
Comment (bms63) [inline posts/admiral/valuelevel/derive_param_computed.qmd:53]: I think we can remove this line

File: posts/admiral/valuelevel/derive_param_computed.qmd [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+$$
+
+``` r
+# read in data
+path_ads <- ""
+adex <- haven::read_sas(paste0(path_ads, "adex.sas7bdat")) %>% 
+  select(USUBJID, PARAM, PARAMCD, AVAL)
+
+# subset input data to PARAMCD = "TOTDOSE", "NUMCYC"
+adex_TOTDOSE_NUMCYC <- adex %>% filter(PARAMCD %in% c("TOTDOSE", "NUMCYC"))
```
Comment (bms63) [inline posts/admiral/valuelevel/derive_param_computed.qmd:52]:
```suggestion
adex_TOTDOSE_NUMCYC <- adex %>% 
filter(PARAMCD %in% c("TOTDOSE", "NUMCYC"))
```

File: posts/admiral/valuelevel/derive_param_computed.qmd (hunk 1 above)
Comment (kaz462) [reply posts/admiral/valuelevel/derive_param_computed.qmd]: links are added

File: posts/admiral/valuelevel/derive_param_computed.qmd (hunk 3 above)
Comment (kaz462) [reply posts/admiral/valuelevel/derive_param_computed.qmd]: Added a brief intro at the last section, let me know if further explanation needed. Thanks for the review!
> For validation purpose, the [diffdf](https://gowerc.github.io/diffdf/reference/diffdf.html) package is used below to mimic SAS proc compare.

**Comment by bms63** [comment]
Hi @kaz462.  I did a few little polishes on the post.  If you are okay with them, then we can merge them into `main`

**Pull request merged**